### PR TITLE
WIP: Baremetal Provisioning Libvirt In a Pod

### PR DESCRIPTION
**Pull request by dgoodwin**

### CODE REVIEW

File: pkg/install/generate.go [hunk 1]
```diff
@@ -291,11 +291,45 @@ func InstallerPodSpec(
...
+							// host
+							Value: "16509",
+						},
+					},
+				*/
+			},
+		)
+		// TODO: host networking is used to access the networks for target clusters.
+		// instead we should provide access: https://docs.openshift.com/container-platform/4.2/networking/multiple-networks/attaching-pod.html
+		hostNetwork := true
```
Comment (staebler) [inline pkg/install/generate.go]: `hostNetwork = true`

File: pkg/install/generate.go [hunk 2]
```diff
@@ -291,11 +291,45 @@ func InstallerPodSpec(
...
 	}
 
+	var hostNetwork bool
+	if cd.Spec.Platform.BareMetal != nil {
+		containers = append(containers,
+			corev1.Container{
+				Name: "libvirt",
+				// TODO: need an official image for this
+				Image:           "docker.io/djzager/libvirt:netstart",
+				ImagePullPolicy: corev1.PullAlways,
```
Comment (joelddiaz) [inline pkg/install/generate.go]: do we always want 'PullAlways'? or is this temporary?

Comment (dgoodwin) [reply pkg/install/generate.go]: Temporary, not sure what the longterm image we use here will be, although I think we usually use PullAlways don't we?

Comment (joelddiaz) [reply pkg/install/generate.go]: i was just thinking if whatever container repo that houses these images becomes inaccessible, does that mean that installations grind to a halt even though the image might be cached?

Comment (dgoodwin) [reply pkg/install/generate.go]: Could be a problem, but across the board as I believe we're using this for almost everything.

Comment (dgoodwin) [reply pkg/install/generate.go]: Seems like we want this for development setups so we always get our latest image, but in prod/stage we use hashes so IfNotPresent would work fine there. We should probably make this configurable somewhere, HiveConfig perhaps.

**Comment by dgoodwin** [comment]
This work might be dead for time being, and at very least we need to restore the ssh approach I removed in here. Closing for now and focusing on solidifying the ssh libvirt approach.

**Pull request closed without merging**